p2p/peer.py

```diff
@@ -342,7 +343,81 @@ def is_closing(self) -> bool:
     async def _cleanup(self) -> None:
         self.close()
 
+    async def boot(self) -> None:
+        try:
+            await self._boot()
+        except OperationCancelled:
+            # If a cancellation happens during boot we suppress it here and
+            # simply exit without setting the `booted` event.
+            return
+        else:
+            self.booted.set()
+
+    async def _boot(self) -> None:
+        try:
+            await self.ensure_same_side_on_dao_fork()
+        except DAOForkCheckFailure as err:
+            self.logger.debug("DAO fork check with %s failed: %s", self, err)
+            await self.disconnect(DisconnectReason.useless_peer)
+            raise
+        except Exception as err:
+            self.logger.exception('ERROR BOOTING')
+            raise
+
+    vm_configuration: Tuple[Tuple[int, Type[BaseVM]], ...] = None
+
+    async def ensure_same_side_on_dao_fork(self) -> None:
```

Comment:
With https://github.com/ethereum/py-evm/pull/1265/files#r216126130, this whole function could live outside of `BasePeer` to make it more generic.